doc/elements-tx-format.md

```diff
@@ -0,0 +1,570 @@
+# Elements Transaction Format
+
+This document describes the format used to serialize elements transactions. Once a transaction has been converted into this raw, serialized form, it can be broadcast across the network.
+
+This document requires familiarity with the architecture of bitcoin and elements, as well the various parts that are expected in an elements transaction (inputs, outputs, assets, pegin data, etc.). For more information on those, please refer to the [elements code tutorial](https://elementsproject.org/elements-code-tutorial/overview).
+
+### Data Types
+
+*Note:* The encoding column in the following tables corresponds to the way that the fields are serialized. In bitcoin, numbers and hashes are serialized in little endian, and this same notion carries over to elements.
+
+*Note:* Values are listed in multiples of 100,000,000 (this corresponds directly to how bitcoin encodes value, where 1 BTC is divisible into 100,000,000 satoshis). For example, a `ConfidentialAmount` representing 7 inflation keys would be encoded as 700,000,000.
+
+#### Transaction
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Version | Yes | 4 bytes | `UnsignedInteger` | Little endian | Protocol version (should always be 2). |
+| Flags | Yes | 1 byte | | | Whether the transaction contains a witness. Either 0 or 1. |
+| Num Inputs | Yes | Varies | `VarInt` | | Number of inputs to the transaction. |
```

Comment:
I sort of agree with you... however, I think that noting this might create more confusion.

For one, VarInts also have their own specific encoding (little-endian).
Additionally, we can't state any other concrete type (such as `uint32`) because we don't know the integer's size in advance.